### DISCUSSION TITLE
Welcome to pyShelf Discussions!

### DISCUSSION THREAD

**Discussion by th3r00t** [th3r00t/pyShelf#65] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Normanras** [comment]
hey th3r00t! Found you out of my relentless search for better ebook self hosting. I am currently running calibre-web on an arm device and can't run calibredb or content server. I haven't tried ubooquity since development seems to have lagged. This brought me to you!

What do you need help with? I am not a professional developer but I can figure my way around some code. Let me know if you need help.

**Reply by th3r00t** [reply]
Hey thanks for reaching out, I'm gonna be digging in here over the next couple of days to see what needs polishing before moving on to new feature implementation.

As for what I could use some help on that all depends on where you're comfortable working. If you prefer html and ui work both the mobile site and admin site could use a lot of help lol.

If python is more your fortay the websocket implementation could use help in the realm of more error checking and recovery, specifically when first launched sometimes the websocket server either is not ready or doesn't reset properly.

Also I've found a bug recently in the docker deployment where the ws server is either timing out or isn't responding at all when attempting an import.

**Reply by Normanras** [reply]
Definitely Python, but happy to help on the front end too. Let me start taking a look at the websockets
